packages/components/src/renderer/FileMessage/index.tsx

```diff
@@ -47,6 +53,13 @@ export const Image: React.FC<MessageTypeHandlerProps<'image'>> = ({ image, title
   )
 }
 
-const useMimeType = (url: string): string | undefined => {
-  return useMemo(() => mimeTypes.getType(path.extname(url)) || undefined, [url])
+const useMimeType = (url: string): string | null => {
+  return useMemo(() => {
+    try {
+      const validUrl = new URL(url)
+      return mimeTypes.getType(validUrl.pathname)
+    } catch (error) {
+      return null
+    }
```

Comment:
```suggestion
    let extension = ''
    try {
      const validUrl = new URL(url)
      extension = validUrl.pathname
    } catch (error) {
      extension = path.extname(url)
    }
    
    return mimeTypes.getType(extension)
```

My bad, I forgot that we had to push a fix for the PR I linked (fix: https://github.com/botpress/botpress/pull/5488). 

The result is a blend of using `new URL` to try to decode the URL and then find the file's extension. In case the URL is not valid (e.g. missing an http://) but still contains an extension, we then use path.extname() to extract it.

Reply:
fixed